R/utils.R

```diff
@@ -189,25 +189,46 @@ check_filter_datanames <- function(filters, datanames) {
 #' @param modules (`teal_modules`) object
 #' @param filters (`teal_slices`) object
 #' @param filtered_data_singleton A result of `teal_data_to_filtered_data` applied to `data`.
+#' @param progress (`Progress`) object, optional; will be filled
 #' @return Returns list of same shape as `modules`, containing `FilteredData` at every leaf.
 #' If module specific, each leaf contains different instance, otherwise every leaf contains `filtered_data_singleton`.
 #' @keywords internal
-modules_datasets <- function(data, modules, filters, filtered_data_singleton = teal_data_to_filtered_data(data)) {
+modules_datasets <- function(data,
+                             modules,
+                             filters,
+                             filtered_data_singleton = teal_data_to_filtered_data(data),
+                             progress) {
   checkmate::assert_class(data, "teal_data")
   checkmate::assert_multi_class(modules, c("teal_modules", "teal_module"))
   checkmate::assert_class(filters, "modules_teal_slices")
   checkmate::assert_r6(filtered_data_singleton, "FilteredData")
+  if (!missing(progress)) checkmate::assert_r6(progress, "Progress")
 
   if (!isTRUE(attr(filters, "module_specific"))) {
+    if (!missing(progress)) {
+      progress$inc(
+        amount = progress$getMax(),
+        detail = sprintf("%.0f%%", progress$getMax() / progress$getMax() * 100)
+      )
+    }
+
     # subset global filters
     slices <- shiny::isolate({
       Filter(function(x) x$id %in% attr(filters, "mapping")$global_filters, filters)
     })
     filtered_data_singleton$set_filter_state(slices)
+
     return(modules_structure(modules, filtered_data_singleton))
   }
 
   if (inherits(modules, "teal_module")) {
+    if (!missing(progress)) {
+      progress$inc(
+        amount = 1,
+        detail = sprintf("%.0f%%", progress$getValue() / progress$getMax() * 100)
```

Comment:
maybe instead of `"%.0f%%"` we can use `round()`? 

```r
sprintf("%s", round(progress$getValue() / progress$getMax() * 100, 2))
```

Reply:
`sprintf` is used throughout the code base, I though it would be more consistent.

Reply:
Yeah. `sprintf` with `%s` not`%.0f%%`. `sprintf`  is fine : ) I meant it's argument

Reply:
Ok then.